src/main/java/org/spongepowered/common/mixin/core/network/MixinNetHandlerPlayServer.java

```diff
@@ -714,4 +715,14 @@ public void onProcessAnimationHead(CPacketAnimation packetIn, CallbackInfo ci) {
     public void setLastMoveLocation(Location<World> location) {
         this.lastMoveLocation = location;
     }
+
+    @Inject(method = "kickPlayerFromServer", at = @At("HEAD"))
+    public void onKickPlayer(String reason, CallbackInfo info) {
+        final MessageChannel originalChannel = ((Player) this.playerEntity).getMessageChannel();
+        final KickPlayerEvent event = SpongeEventFactory.createKickPlayerEvent(
+                Cause.of(NamedCause.source(this.playerEntity)), SpongeTexts.fromLegacy(reason), (Player) this.playerEntity
+        );
+
+        reason = SpongeTexts.toLegacy(event.getReason());
```

Comment:
You can't reassign variables like this.
There are multiple possible ways of doing this though: `@Redirect`, `@ModifyVariable` `@ModifyArg`